### DISCUSSION TITLE
FadCam v1.2.1-Beta

### DISCUSSION THREAD

**Discussion by anonfaded** [anonfaded/FadCam#67] [Announcements]
**✨ What's New?**

**🚀 New Features:**  
- 💡 Flashlight functionality added.

**🔧 Improvements:**  
- ❌ Fixed crash when selected framerate was incompatible with the camera.  
- 🔄 Fixed "Check for Updates" button crashing in beta versions.

**👥 Contributors:**  
- 🛠️ **@s0llvan**: Contributions in https://github.com/anonfaded/FadCam/issues/63 and https://github.com/anonfaded/FadCam/issues/61

<hr /><em>This discussion was created from the release <a href='https://github.com/anonfaded/FadCam/releases/tag/v1.2.1-beta'>FadCam v1.2.1-Beta</a>.</em>

**Comment by dlinkdns** [comment]
Thank you for your amazing work. I am a fan. Just wanted to let you know i have a oppo reno android version 12. Since i updated to the latest version 12.01 beta the app now doesn't load. Unfortunately i cant go back and i have loads of videos

**Reply by anonfaded** [reply]
> Thank you for your amazing work. I am a fan. Just wanted to let you know i have a oppo reno android version 12. Since i updated to the latest version 12.01 beta the app now doesn't load. Unfortunately i cant go back and i have loads of videos

Thank you for your kind words:)
Regarding the videos, please backup your data by accessing the videos from google file manager application. Go to this location to find all the videos: `Android/data/com.fadcam/files/fadcam` and copy the videos to another folder liken `~/Downloads` for example. But if you uninstalled the app and then installed the beta version, which is the only way to have updated version btw, then all the files will be gone as the internal storage got deleted with the app as well. Sad if this is the case for you:(

Also why the app doesn't load? Like there isn't any such major refactoring then i wonder why this happens. Explain the whole bug please. Also i would suggest to clear the cache of app and then try opening. There are 1k+ downloads for this beta version but i haven't heard about this yet happening with anyone.

**Reply by dlinkdns** [reply]
Thank you for your quick reply. Absolutely amazing support!

Luckily, even though I installed the beta version, I didn’t lose the data. I followed your advice and successfully backed up the folder. I was initially struggling to view the files as they were not displaying, but a quick Google search suggested installing the "Files" app. After doing so, I managed to back up everything.

I also cleared the cache, but unfortunately, it didn’t resolve the issue.

Here’s what happens:

When I load the app, the screen briefly goes white for a split second and then returns to the mobile desktop screen.

I have a question now: If I uninstall and reinstall the beta or full version, I understand that I will lose everything in the data folder. If I copy the fadcam folder back to the location Android/data/com.fadcam/files/fadcam, will the app be able to recognize the files and display them again?

Thank you for your help!

**Reply by anonfaded** [reply]
> I didn’t lose the data.

Good to hear that!



> will the app be able to recognize the files and display them again?

I tried on my phone and manually i can't `write` to the internal location, even the `Android/data` location isn't visible when i try to copy a video so i think this folder is read only and we can only take the videos from it and can't add videos from other location. But you can try by copying any video to `Android/data/com.fadcam/files/fadcam` and if you can see the `/data` folder during copying, then the answer is Yes, otherwise that isn't possible i guess. Later it is planned to add a custom folder location to save videos in a location that user decides, but it will take time because I'm not free for some time.



> When I load the app, the screen briefly goes white for a split second and then returns to the mobile desktop screen.

This is weird and right now the app has no crash-logging feature so I'm not able to understand the actual reason of why this happens. Maybe you downloaded the `armeabi` or `arm64` apk and either of them wasn't compatible with your phone? Also try installing the universal apk version, it has all the architectures together which makes its size bulky but it may work. Let me know if it works out or not:)

Optionally you can also join our discord group for faster assistance, Thanks.

**Reply by dlinkdns** [reply]
Thank you for your reply.

I managed to resolve the issue by renaming the folder com.fadcam to com.fadcamOLD and then clearing both the cache and storage data. After doing this, the beta version of the app loaded fine.

I then deleted the newly created com.fadcam folder and renamed the original com.fadcamOLD folder back to com.fadcam. All the videos are now back and accessible within the app.

Thank you again for your help—perhaps this information will assist someone else in the future!

Best regards,

**Reply by anonfaded** [reply]
Oh, great! I hadn't thought about this. I'll remember it and share it with other users if someone mentions this issue again.

**Comment by tazztone** [comment]
hi great app thanks!
i wish we could select 5x back lens for video recording. would it be possible to implement ?
cheers

**Reply by anonfaded** [reply]
Yeah i saw your review as well mentioning this. Yes it is possible, and is already planned as some other users also mentioned this, but it would need a lot of testing. If you are willing to then you can join our discord group and whenever this is implemented, you can test and confirm as i don't have a device with more cameras so it would be a guess work only.
